container/snapshot.go

```diff
@@ -0,0 +1,152 @@
+package container
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/Sirupsen/logrus"
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/network"
+	"github.com/docker/go-connections/nat"
+)
+
+// Snapshot is a read only view for Containers
+type Snapshot struct {
+	ID           string `json:"Id"`
+	Name         string
+	Pid          int
+	Managed      bool
+	Image        string
+	ImageID      string
+	Command      string
+	Ports        []types.Port
+	ExposedPorts nat.PortSet
+	PublishPorts nat.PortSet
+	Labels       map[string]string
+	State        string
+	Status       string
+	Health       string
+	HostConfig   struct {
+		NetworkMode string
+		Isolation   string
+	}
+	NetworkSettings types.SummaryNetworkSettings
+	Mounts          []types.MountPoint
+	Created         time.Time
+	StartedAt       time.Time
+	Running         bool
+	Paused          bool
+	ExitCode        int
+}
+
+// Snapshot provides a read only view of a Container. Callers must hold a Lock on the container object.
+func (container *Container) Snapshot() *Snapshot {
```

Comment:
I started with that, but it had a pointer to `*NetworkSettings` and was missing a few fields that `daemon/list.go` requires. Since `types.Container` is a public API, I didn't want break API compatibility or mess with its JSON serialization.

It's probably possible, but a bit riskier I think.